crates/sui-framework-build/src/unit_tests/build_tests.rs

```diff
@@ -22,3 +25,41 @@ fn generate_struct_layouts() {
         registry.contains_key("0000000000000000000000000000000000000002::tx_context::TxContext")
     );
 }
+
+#[test]
+fn generate_lock_file() {
+    let mut path = Path::new(env!("CARGO_MANIFEST_DIR"))
```

Comment:
Let's not write the lock file to `sui/crates/sui-framework`, because if the test is interrupted for some reason, it means that we will pollute the repo, which is not ideal.  Given we have the flexibility to change the lock file path, we should use it here to write it out to a `TempFile` (see the `tempfile` crate).

Also, the lock file for `sui-framework` is uniquely boring, so I would suggest moving the test to a place where you can try to build a more interesting package.  `move_integration_tests` is your best bet for that, I think, because you will still have the flexibility to update the `build_config` to set the lock file there.